apps/sms/js/compose.js

```diff
@@ -741,46 +741,64 @@ var Compose = (function() {
-        AttachmentMenu.open(this.currentAttachment);
+      if (!event.target.classList.contains(attachmentClass) || state.resizing) {
+        return;
       }
-    },
 
-    onAttachmentMenuClick: function thui_onAttachmentMenuClick(event) {
-      event.preventDefault();
-      switch (event.target.id) {
-        case 'attachment-options-view':
-          this.currentAttachment.view();
-          break;
-        case 'attachment-options-remove':
-          dom.message.removeChild(this.currentAttachmentDOM);
-          disposeAttachmentNode(this.currentAttachmentDOM);
+      this.currentAttachmentDOM = event.target;
+      this.currentAttachment = attachments.get(event.target);
 
-          state.size = null;
-          onContentChanged();
-          AttachmentMenu.close();
-          break;
-        case 'attachment-options-replace':
-          var request = this.requestAttachment();
-          request.onsuccess = (function replaceAttachmentWith(newAttachment) {
-            var fragment = insert(newAttachment);
+      var name = this.currentAttachment.name;
+      var blob = this.currentAttachment.blob;
+      var options = {
+        header: name.substr(name.lastIndexOf('/') + 1)
+      };
 
-            dom.message.insertBefore(fragment, this.currentAttachmentDOM);
+      // Localize the name of the file type
+      var mimeFirstPart = blob.type.substr(0, blob.type.indexOf('/'));
+      var fileType = (TYPES.indexOf(mimeFirstPart) > -1) ?
+        mimeFirstPart: 'others';
 
+      options.items = [
+        {
+          l10nId: 'view-attachment-' + fileType,
+          method: () => {
+            this.currentAttachment.view();
+          }
+        },
+        {
+          l10nId: 'remove-attachment-' + fileType,
+          method: () => {
             dom.message.removeChild(this.currentAttachmentDOM);
             disposeAttachmentNode(this.currentAttachmentDOM);
 
-            onContentChanged(newAttachment);
-            AttachmentMenu.close();
-          }).bind(this);
-          request.onerror = this._onAttachmentRequestError;
-          break;
-        case 'attachment-options-cancel':
-          AttachmentMenu.close();
-          break;
-      }
+            state.size = null;
+            onContentChanged();
+          }
+        },
+        {
+          l10nId: 'replace-attachment-' + fileType,
+          method: () => {
+            var request = this.requestAttachment();
+            request.onsuccess = (function replaceAttachmentWith(newAttachment) {
+              var fragment = insert(newAttachment);
+
+              dom.message.insertBefore(fragment, this.currentAttachmentDOM);
+
+              dom.message.removeChild(this.currentAttachmentDOM);
+              disposeAttachmentNode(this.currentAttachmentDOM);
+
+              onContentChanged(newAttachment);
+            }).bind(this);
+            request.onerror = this._onAttachmentRequestError;
+          }
+        },
```

Comment:
nit: I think it would be more readable to define the various functions before defining the parameter object.